### PR TITLE
refactor(clerk-js): Improve avatar save UX

### PR DESCRIPTION
**Pull request by alexcarpenter**
### Issue

Folks find it very confusing that uploading an avatar autosaves it, and doesn't allow hitting the save button, which requires them to hit cancel after it uploads to close the form.

### Solution

Defer saving the image until the user clicks save which will close the form, and allow canceling which reverts the upload.

https://github.com/user-attachments/assets/a49f485c-0bc1-49c4-b52c-72acbb14b421

> [!NOTE]
> When removing the avatar, we should the initials fallback until its saved, then it renders the gradient avatar that is generated. Not sure we'd have access to the gradient avatar until the form is saved after removing.

## Checklist

- [ ] `pnpm test` runs as expected.
- [ ] `pnpm build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Summary by CodeRabbit

* **New Features**
  * Avatar upload/removal is deferred until you click Save; live preview, reset, and same-file re-selection supported.

* **Bug Fixes**
  * Prevents empty image sources, ensures preview URLs are cleaned up, and shows the remove action consistently.

* **Tests**
  * Expanded unit and integration tests for upload, remove, reset, save, cancel, and edge cases.

* **Chores**
  * Updated UX messaging to reflect deferred avatar upload behavior.

### CODE REVIEW

File: packages/clerk-js/src/ui/elements/AvatarUploader.tsx [hunk 1]
```diff
@@ -105,14 +103,15 @@ export const AvatarUploader = (props: AvatarUploaderProps) => {
...
-            {!!onAvatarRemove && !showUpload && (
+            {!!onAvatarRemove && (
               <Button
                 elementDescriptor={descriptors.avatarImageActionsRemove}
                 localizationKey={localizationKeys('userProfile.profilePage.imageFormDestructiveActionSubtitle')}
                 isDisabled={card.isLoading}
                 sx={t => ({ color: t.colors.$danger500 })}
                 variant='ghost'
                 colorScheme='danger'
+                size='xs'
```
Comment (alexcarpenter) [inline packages/clerk-js/src/ui/elements/AvatarUploader.tsx:114]: Fixing a height shift when the remove button is conditionally rendered. Matches the upload button `size` usage.
